### DISCUSSION TITLE
VSCode Intellisense is not suggesting available methods in JEST

### DISCUSSION THREAD

**Discussion by kartheektrilochan** [orta/vscode-jest#686] [Q&A]
### Environment
1. `node -v`: v.9.2.0
1. `npm -v`: 5.5.1
1. `npm ls jest` or `npm ls react-scripts` (if you haven’t ejected):
npm ls jest 
+-- jest@24.5.0
`-- react-scripts@2.1.8
  `-- jest@23.6.0

1. your vscode-jest settings if customized:
    - jest.pathToJest? [fill]
    - jest.pathToConfig? [fill]
    - anything else that you think might be relevant? [fill]

1. Operating system: Windows_NT x64 10.0.17134

### Prerequisite
- are you able to run jest test from command line? yes
- how do you run your tests from command line? npm run test

Steps to Reproduce:

Create a new react app using 

1 .npx create-react-app my-app 
2. npm install --save-dev jest
3. Opened the project in VSCode.

 

### Expected Behavior
I expect VS Code intellisense would assist me with available methods from JEST.

### Actual Behavior

VS Code is not suggesting me any method suggestions for JEST test cases.

**Comment by sewbacca** [comment]
I'm facing the same problem. Testing works fine, but methods aren't suggested.
npm -v: 6.5.0
node -v: v11.9.0

**Comment by dlee005c** [comment]
same for me as well.
node -v v9.6.1
npm -v 6.9.0
vscode: version 1.33.1 (1.33.1)
For what it's worth, i'm using nvm to manage node/npm

**Comment by cassiobock** [comment]
I'm having the same problem. I've tried installing [@types/jest](https://www.npmjs.com/package/@types/jest) with `npm i @types/jest` and the suggestions started to show again.

**Comment by filipeestacio** [comment]
I found that I need to restart vscode after installing @types/jest if vscode was open during installation. Can anyone confirm if that is expected?

**Comment by Waserman** [comment]
create `jsconfig.json`  in the root directory of your project.
the file should contain the following:
`
{
    "typeAcquisition": {
        "include": [
            "jest"
        ]
    }
}
`

after saving reload vscode.
Worked for me.

**Comment by redeemefy** [comment]
> create `jsconfig.json` in the root directory of your project.
> the file should contain the following:
> `{ "typeAcquisition": { "include": [ "jest" ] } }`
> 
> after saving reload vscode.
> Worked for me.

For some reason this solution doesn't work for me. Interesting is that I found a post from Nicholas Zakas post with the solution you are proposing. I'm wondering if I need to tell vscode about `jsconfig.json` or doesn vscode just picked it up if its there.

**Comment by noxasch** [comment]
> I'm having the same problem. I've tried installing [@types/jest](https://www.npmjs.com/package/@types/jest) with `npm i @types/jest` and the suggestions started to show again.


mine only works after i install `@types/jest` as suggested

node v10.15.0
npm v6.5.0

```
npm i -D @types/jest
```

**Reply by pratham-soni** [reply]
worked for me. thanks @noxasch

**Comment by danielo515** [comment]
None of the provided solutions works for me. Guess what else can I try

**Comment by redeemefy** [comment]
> None of the provided solutions works for me. Guess what else can I try

Try to create a new user in your computer with a fresh installation of vscode and do `npm i -D @types/jest` and see if that works. I'm just trying to isolate all other extensions that might be causing conflict.

**Comment by nandosalas28** [comment]
Install `@types/jest` as a dev dependencie works for me.
I've tried to install it globally but it seems not work.

**Comment by jjayaraman** [comment]
Thanks, Guys, as mentioned above just the below in jsconfig.json works for me.
I thought in javascript projects this is a better option, as javascript project like ReactJS does not need to install @types/xxxx. I guess @types are only for typescript projects? 

```
{
    "typeAcquisition": {
        "include": ["jest"]
    }
}
```

**Comment by or4** [comment]
install local and global types, works fine
npm i @types/jest
npm i -g @types/jest

**Comment by jcairo** [comment]
Including the test folder in tsconfig fixed it for me.

`...  "include": ["src", "test"]`

**Comment by andmill** [comment]
> 
> 
> Including the test folder in tsconfig fixed it for me.
> 
> `... "include": ["src", "test"]`

This worked for me too. I have the _orta.vscode-jest_ extension installed and am running a react 16 project with `npx create-react-app`.
No restart of VSCode required

**Comment by jcairo** [comment]
Only problem is that it then compiles the code test project folder. Still not a great solution...

**Comment by leaveswoods** [comment]
Installing @types/jest solved the problem while adding this to jsconfig.js could not help`
{
    "typeAcquisition": {
        "include": ["jest"]
    }
}`

**Comment by sumeetchawla** [comment]
> I found that I need to restart vscode after installing @types/jest if vscode was open during installation. Can anyone confirm if that is expected?

Yup, yesterday it wasn't working. I closed and went to sleep and it is working now. I did a lot of window reload yesterday but didn't work either.

**Comment by azad21012** [comment]
before  read this issue i thought  need an extension like jest -published by orta- . but after installing @types/jest and reload vs-code the methods suggested successfully by vs-code :)

**Comment by onuriltan** [comment]
**P.S** Doing both steps worked for me, and the last one 🙂 
1. `npm i -D @types/jest`
2. Add `jsconfig.json` to root folder as
```
{
    "typeAcquisition": {
        "include": ["jest"]
    }
}
```
3. Reload VSCode

**Reply by fabrizioGigli** [reply]
That's a solution. If this doesn't work after this, try to clear VSCode cache:

1. press Ctrl + Shift + P
2. type command Clear Editor History
3. press Enter

It worked well for me and no need to restart VSCode, maybe this can help.

**Comment by jvmazagao** [comment]
> **P.S** Doing both steps worked for me, and the last one 🙂
> 
> 1. `npm i -D @types/jest`
> 2. Add `jsconfig.json` to root folder as
> 
> ```
> {
>     "typeAcquisition": {
>         "include": ["jest"]
>     }
> }
> ```
> 
> 1. Reload VSCode

In my vscode this solution works very well.

**Reply by daguitosama** [reply]
Totally worked out for me too

**Comment by Mr-Kuzmenko-K** [comment]
> Install `@types/jest` as a dev dependencie works for me.
> I've tried to install it globally but it seems not work.

Dude, thank you so much

**Comment by yevhen-logosha** [comment]
my case was an outdated version of `@types/jest`, upgrading it to match version of Jest did the trick

**Comment by dil-dbakker** [comment]
My case was opening VSCode at the root of our repo, rather than the src folder which contained our jest.config.js and tsconfig.json files. When I opened the src folder as a VSCode Project, everything worked as expected.

**Comment by aardvarkk** [comment]
@dil-dbakker Thank you! This was exactly my problem. I tried about a thousand complicated config options people were proposing but I was just one directory "too high" opening a parent directory in VSCode. Running `vscode .` inside the specific project I was working with fixed everything for me.

**Reply by alfonsoruiz** [reply]
This was the only solution that worked for me as well.

**Comment by alonsogchparra** [comment]
> create `jsconfig.json` in the root directory of your project.
> the file should contain the following:
> `{ "typeAcquisition": { "include": [ "jest" ] } }`
> 
> after saving reload vscode.
> Worked for me.

This suggestion worked for me very well. Thanks for the feedback @Waserman

**Comment by connectdotz** [comment]
great thread to help people resolving similar issues. But since this issue is really not related to this extension, I will close it.

**Comment by wbolduc** [comment]
> great thread to help people resolving similar issues. But since this issue is really not related to this extension, I will close it.
Where would you recommend this information goes

I was experiencing this same sort of problem where intellisense would not show the available jest methods despite doing what almost every thread on the topic suggests

I was able to resolve it but I don't know why this works.

_remove_ `{ "typeAcquisition": { "include": [ "jest" ] } }`
make sure that in the jsconfig.json you are not specifying a compilerOption.target

My resulting config looks like this (minus work specific stuff)
`{
	//for some reason type acquisition and and target both break intellisense for jest
	"compilerOptions": {
		"allowSyntheticDefaultImports": false,
		"baseUrl": "./",
		"paths": {
			"Test/*": ["./__mocks__/*"]
		}
	},
	"include": ["./src/**/*"]
}
`

The packages I have installed relevant to testing
`
  "@testing-library/jest-dom": "^5.11.10",
  "@testing-library/react": "^11.2.6",
  "@testing-library/user-event": "^13.1.3",
  "@types/jest": "^26.0.22",
  "@types/react": "^17.0.3",
  "babel-jest": "^26.6.3",
  "babel-plugin-transform-class-properties": "^6.24.1",
  "eslint-plugin-jest": "^24.3.5",
  "jest": "^26.6.3",
  "jest-dom": "^4.0.0",
  "msw": "^0.28.1",
`

I don't totally know why this works but it's worth a shot if you were stuck like me

**Comment by gianluca-sabena** [comment]
Another workaround is to use explicit import: "However, if you prefer explicit imports, you can do import {describe, expect, test} from '@jest/globals'."as documented here https://jestjs.io/docs/api

PS: I was not able to solve the problem with other suggest methods.

**Reply by emekaelo** [reply]
After much research 🥵, this worked for my Angular 13 project.
Thanks

**Reply by tanujveera** [reply]
I have been searching for this solution. This worked for me. Really thanks for your info.

**Comment by dreadwail** [comment]
If none of the above solutions work for you, you may be running into a similar issue as I was.  The problem I was encountering was that type checking worked fine for all the `jest.*` methods, but type checking was not finding types for globals such as `expect()`.

I was running a fairly old version of jest (`26.x`) but had `@types/jest` at an even older version (`24.x`).  Even though the types for `@types/jest` at version `24.x` had global type definitions for `expect` they were not getting picked up no matter what I would do. Even after trying all the advice in this discussion thread it would always resolve `expect` as `any`.

I didn't initially notice the version mismatch between `jest` and `@types/jest` but even when I did I still would have expected the valid global definition for `expect` in `@types/jest` `24.x` to get picked up and used, but it wasn't.

I decided after exhausting all other options to align the version of `@types/jest` to `26.x`.  I didn't think this would make any difference because the global `expect` definition in both looked pretty identical.  Yet, when I upgraded `@types/jest` to `26.x` everything just started working. 😕   This doesn't make much sense to me, but hopefully this helps someone else who finds this thread and is in a similar situation.

**Comment by mrdzick** [comment]
Try to delete `node_modules` folder, and type `npm install` to install all dependencies again. It works for me.

**Comment by PascalUlor** [comment]
This `npm i -D @types/jest` worked for my node project
